R/geom-.R

```diff
@@ -79,7 +81,7 @@ Geom <- ggproto("Geom",
     }
 
     # Trim off extra parameters
-    params <- params[intersect(names(params), self$parameters())]
+    params <- filter_args(params, self$draw_panel)
```

Comment:
4. We then filter the parameters on the go in the `compute/draw_layer()` and `compute/draw_panel()` methods